src/app/qgisappstylesheet.cpp

```diff
@@ -106,18 +108,24 @@ void QgisAppStyleSheet::buildStyleSheet( const QMap<QString, QVariant> &opts )
   if ( fontSize != defaultSize || fontFamily != defaultFamily )
     ss += QStringLiteral( "* { font: %1pt \"%2\"} " ).arg( fontSize, fontFamily );
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 12, 2)
   // Fix for macOS Qt 5.9+, where close boxes do not show on document mode tab bar tabs
-  // See: https://bugreports.qt.io/browse/QTBUG-61092
-  //      https://bugreports.qt.io/browse/QTBUG-61742
+  // See: https://bugreports.qt.io/browse/QTBUG-61092 => fixed in 5.12.2 / 5.14
+  //      https://bugreports.qt.io/browse/QTBUG-61742 => fixed in 5.9.2
   // Setting any stylesheet makes the default close button disappear.
   // Specifically setting a custom close button temporarily works around issue.
-  // TODO: Remove when regression is fixed (Qt 5.9.3 or 5.10?); though hard to tell,
-  //       since we are overriding the default close button image now.
   if ( mMacStyle )
   {
     ss += QLatin1String( "QTabBar::close-button{ image: url(:/images/themes/default/mIconCloseTab.svg); }" );
     ss += QLatin1String( "QTabBar::close-button:hover{ image: url(:/images/themes/default/mIconCloseTabHover.svg); }" );
   }
+#endif
+  if ( mMacStyle )
```

Comment:
@3nids , have you tried non-default themes (night mapping and/or blend of gray) to make sure this unconditional stylesheet addition doesn't mess those up. 99.9% it won't, but I don't have a mac at-hand to confirm.

Reply:
yes it does look good